cdm/core/src/main/java/ucar/nc2/dataset/VariableDS.java

```diff
@@ -609,172 +612,211 @@ public void removeCoordinateSystem(CoordinateSystem cs) {
 
   @Override
   public boolean hasScaleOffset() {
-    return scaleMissingUnsignedProxy.hasScaleOffset();
+    return scaleOffset != null;
   }
 
   @Override
   public double getScaleFactor() {
-    return scaleMissingUnsignedProxy.getScaleFactor();
+    return scaleOffset != null ? scaleOffset.getScaleFactor() : 1;
   }
 
   @Override
   public double getOffset() {
-    return scaleMissingUnsignedProxy.getOffset();
+    return scaleOffset != null ? scaleOffset.getOffset() : 0;
   }
 
   @Override
   public boolean hasMissing() {
-    return scaleMissingUnsignedProxy.hasMissing();
+    return convertMissing != null ? convertMissing.hasMissing() : false;
```

Comment:
This and several of the methods below that return booleans can be simplified like:
```suggestion
    return convertMissing != null && convertMissing.hasMissing()
```